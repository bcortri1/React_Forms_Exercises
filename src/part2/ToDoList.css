.ToDoList {
    display: flex;
    background-color: white;
    justify-content: space-evenly;

    flex-wrap: wrap;
    flex-flow: column;
    width: 40%;
    margin: auto;
}