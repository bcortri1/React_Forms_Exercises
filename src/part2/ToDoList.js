import React, { useState } from 'react';
import { v4 as uuid } from 'uuid';
import ToDo from './ToDo';
import NewToDoForm from './NewToDoForm';
import './ToDoList.css'


const ToDoList = () => {
    const initialState = []
    const [toDoArr, setToDoList] = useState(initialState);

    const removeToDo = (id) => {
        setToDoList(toDoArr.filter((todo) => (todo.id !== id)));
    }

    const addToDo = (data) => {
        let id = uuid();
        setToDoList([...toDoArr, {...data, id}]);
    }

    return <div>
        <div className='ToDoList'>
            <NewToDoForm addToDo={addToDo}/>
        </div>
        <ol className='ToDoList'>
            {toDoArr.map((props) =>
                <ToDo key={props.id} {...props} removeToDo={removeToDo} />
            )}
        </ol>
    </div>;
};

export default ToDoList;