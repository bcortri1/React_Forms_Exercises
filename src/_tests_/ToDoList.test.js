import React from "react";
import { render, fireEvent } from "@testing-library/react";
import '@testing-library/jest-dom';
import ToDoList from "../part2/ToDoList";



// smoke test
it("renders without crashing", function () {
    render(<ToDoList />);
});

// snapshot test
it("matches snapshot", function () {
    const { asFragment } = render(<ToDoList />);
    expect(asFragment()).toMatchSnapshot();
});


it("add new ToDos", function () {
    const {queryByText} = render(<ToDoList/>)
    const addButton = queryByText("Add!");
    fireEvent.click(addButton);

    expect(document.getElementsByClassName("ToDo")[0]).toBeInTheDocument();
    expect(document.getElementsByClassName("ToDo")[1]).not.toBeDefined();

});

it("can add and contain multiple ToDos", function () {
    const { queryByText, getByLabelText } = render(<ToDoList />)
    const input = getByLabelText("ToDo:");
    const addButton = queryByText("Add!");

    fireEvent.click(addButton);
    expect(document.getElementsByClassName("ToDo")[0]).toBeInTheDocument();

    fireEvent.change(input, { target: { value: "This is a test" } });
    fireEvent.click(addButton);
    expect(document.getElementsByClassName("ToDo")[1]).toBeInTheDocument();
    expect(document.getElementsByClassName("ToDo")[1].innerHTML).toMatch("This is a test");
    expect(document.getElementsByClassName("ToDo")[2]).not.toBeDefined();

});
