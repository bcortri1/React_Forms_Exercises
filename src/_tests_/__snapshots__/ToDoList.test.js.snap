// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="ToDoList"
    >
      <form
        id="new-todo-form"
      >
        <label
          for="text"
        >
          ToDo:
        </label>
        <input
          id="text"
          name="text"
          required=""
          type="text"
          value=""
        />
        <button
          id="form-btn"
        >
          Add!
        </button>
      </form>
    </div>
    <ol
      class="ToDoList"
    />
  </div>
</DocumentFragment>
`;
